engine/src/test/java/io/zeebe/engine/job/JobTimeOutTest.java

```diff
@@ -15,58 +15,49 @@
  * You should have received a copy of the GNU Affero General Public License
  * along with this program.  If not, see <http://www.gnu.org/licenses/>.
  */
-package io.zeebe.broker.engine.job;
+package io.zeebe.engine.job;
 
 import static io.zeebe.protocol.intent.JobIntent.ACTIVATED;
 import static io.zeebe.protocol.intent.JobIntent.COMPLETED;
 import static io.zeebe.protocol.intent.JobIntent.FAILED;
 import static io.zeebe.protocol.intent.JobIntent.TIME_OUT;
+import static io.zeebe.test.util.record.RecordingExporter.jobRecords;
 import static org.assertj.core.api.Assertions.assertThat;
 
-import io.zeebe.broker.test.EmbeddedBrokerRule;
 import io.zeebe.engine.processor.workflow.job.JobTimeoutTrigger;
+import io.zeebe.engine.util.EngineRule;
 import io.zeebe.exporter.api.record.Record;
 import io.zeebe.exporter.api.record.RecordMetadata;
+import io.zeebe.exporter.api.record.value.JobBatchRecordValue;
 import io.zeebe.exporter.api.record.value.JobRecordValue;
-import io.zeebe.protocol.intent.Intent;
 import io.zeebe.protocol.intent.JobBatchIntent;
 import io.zeebe.protocol.intent.JobIntent;
-import io.zeebe.test.broker.protocol.commandapi.CommandApiRule;
-import io.zeebe.test.broker.protocol.commandapi.PartitionTestClient;
+import io.zeebe.test.util.record.RecordingExporter;
 import java.time.Duration;
 import java.util.List;
 import java.util.stream.Collectors;
-import org.junit.Before;
 import org.junit.Rule;
 import org.junit.Test;
-import org.junit.rules.RuleChain;
 
 public class JobTimeOutTest {
-  public EmbeddedBrokerRule brokerRule = new EmbeddedBrokerRule();
-  public CommandApiRule apiRule = new CommandApiRule(brokerRule::getAtomix);
+  private static final String PROCESS_ID = "process";
 
-  @Rule public RuleChain ruleChain = RuleChain.outerRule(brokerRule).around(apiRule);
-  private PartitionTestClient client;
-
-  @Before
-  public void setup() {
-    client = apiRule.partitionClient();
-  }
+  @Rule public EngineRule engineRule = new EngineRule();
 
   @Test
   public void shouldNotTimeOutIfDeadlineNotExceeded() {
     // given
-    brokerRule.getClock().pinCurrentTime();
+    engineRule.getClock().pinCurrentTime();
```

Comment:
Please let us not pin the time, let us just avoid this. I see no benefit in this

Reply:
Currently I need some way to test that the deadline is correct and I think for that I need to pin time. Something like this:
```
    final ControlledActorClock clock = engineRule.getClock();
    clock.pinCurrentTime();

    final Duration timeout = Duration.ofMinutes(4);
    final Instant deadline = clock.getCurrentTime().plus(timeout);
```

And then I can check the record against this deadline. Do you know if I can do this in another way without pinning and exposing the clock or do I have to remove this assertion?

Reply:
First of all this test tries to test again something which should not happen.

Reply:
And each record has a timestamp which you could use

Reply:
Ok my bad, this is the JobTimeOutTest and I was looking at the ActivateJobsTest. In the activate test, I have the setup I described above and then I assert that the record's deadline is correct. How can I test the activated record without using the clock?